dotCMS/src/main/java/com/dotcms/filters/NormalizationFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.dotcms.filters;
+
+import java.io.IOException;
+import java.net.URI;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+/**
+ * Filter created to wrap all the incoming requests to override the {@link
+ * HttpServletRequest#getRequestURI()} method in order to normalize the requested URIs.
+ */
+public class NormalizationFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'filterConfig' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4798850)